File: Sources/hrt.dal/src/PositionCandidateStepDal.js

const { SQLDal } = require('./SQLDal');
const PositionCandidateStepEntity = require('./entities/PositionCandidateStep');
const mssql = require('mssql');

class PositionCandidateStepDal extends SQLDal {

    init(initParams) {
        super.initDal(initParams);        
    }

    async GetAll() {
        let statuses = null;
        await mssql.connect(this._config);

        let records = await super.execStorProcRecordset(mssql, 'p_PositionCandidateStep_GetAll');

        if(records) {
            statuses = [];
            for(let r in records) {
                let step = new PositionCandidateStepEntity();
                step.StepID = parseInt(records[r].StepID);
                step.Name = records[r].Name;
                step.ReqDueDate = records[r].ReqDueDate;
                step.RequiresRespInDays = records[r].RequiresRespInDays;

                statuses.push(step);
            }
        }

        return statuses;
    }
}

module.exports = PositionCandidateStepDal;

File: Sources/HiringTracker.API/HRT.DAL.MSSQL/PositionCandidateStepDal.cs
using HRT.Common;
using HRT.Interfaces;
using HRT.Interfaces.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.Composition;
using System.Data;
using System.Text;

namespace HRT.DAL.MSSQL
{
    class PositionCandidateStepDalInitParams : InitParamsImpl
    {
    }

    [Export("MSSQL", typeof(IPositionCandidateStepDal))]
    public class PositionCandidateStepDal : SQLDal, IPositionCandidateStepDal
    {
        public IInitParams CreateInitParams()
        {
            return new PositionCandidateStepDalInitParams();
        }

        public bool Delete(long id)
        {
            throw new NotImplementedException();
        }

        public PositionCandidateStep Get(long id)
        {
            throw new NotImplementedException();
        }

        public IList<PositionCandidateStep> GetAll()
        {
            IList<PositionCandidateStep> result = base.GetAll<PositionCandidateStep>("p_PositionCandidateStep_GetAll", PositionCandidateStepFromRow);

            return result;
        }

        public void Init(IInitParams initParams)
        {
            InitDbConnection(initParams.Parameters["ConnectionString"]);
        }

        public long? Upsert(PositionCandidateStep entity, long? editorID)
        {
            throw new NotImplementedException();
        }

        public PositionCandidateStep Upsert(PositionCandidateStep entity)
        {
            throw new NotImplementedException();
        }

        #region Support method
        private PositionCandidateStep PositionCandidateStepFromRow(DataRow row)
        {
            var entity = new PositionCandidateStep();

            entity.Name = (string)row["Name"];
            entity.StepID = (long)row["StepID"];
            entity.ReqDueDate = (bool)row["ReqDueDate"];
            entity.RequiresRespInDays = (int)row["RequiresRespInDays"];

            return entity;
        }
        #endregion
    }
}
